src/ts/components/uicontainer.ts

```diff
@@ -49,6 +49,9 @@ export class UIContainer extends Container<UIContainerConfig> {
   private userInteractionEventSource: DOM;
   private userInteractionEvents: { name: string, handler: EventListenerOrEventListenerObject }[];
 
+  public hideUi: () => void = () => {};
+  public showUi: () => void = () => {};
```

Comment:
I had a closer look at what it means to make those two methods public.

_IMO the usage of this feels weird._

### Why?

In order to create a UI it's very handy to just use our provided Factories (same as we do in `index.html`)

Create the UI:
```
    // Add UI
    uiManager = bitmovin.playerui.UIFactory.buildDefaultUI(player, uiConfig);
```

So now a user has this `uiManager` at hand. One would expect to be able to fully toggle the UI visible. So where would you look for this? I would assume on the `uiManager` as this is the thing to interact with.

Though there is nothing :travolta:

Instead, you need to know the internals of our UI and how it's setup (multiple UI variants inside one `uimanager`)

Ok how to access those UIContainer instances now 🤔  let's search for them 💪 The only thing you find will be the `uiManager.getUiVariants()` so you would need to do `uiManager.getUiVariants().forEach((variant) => variant.ui.hideUi())` which is super inconvenient.

---

I would propose to now expose them now or add something on the `uiManager` to enable a full visibility toggle hiding all the `uiVariant` complexity.